libraries/eosiolib/contracts/eosio/map.hpp

```diff
@@ -298,7 +298,7 @@ namespace eosio::kv {
 
          inline bool operator!=(const iterator& o) const { return !((*this) == o); }
 
-         void materialize() {
+         void materialize() const {
```

Comment:
Notes for future versions:
- this will re-materialize the element that you may hold a reference to but only when you dereference the iterator you got it from.  This seems like an odd interaction from the outside.
- perhaps `element` should be a `std::unique_ptr` or other such construct to reduce the size of an iterator when the value is large and to enforce less requirements (copiability/movability/default-constructability etc) on the value type